src/sklearndf/pipeline/wrapper/_wrapper.py

```diff
@@ -213,7 +219,25 @@ def _more_tags(self) -> Dict[str, Any]:
         )
 
 
-class FeatureUnionWrapperDF(TransformerWrapperDF[FeatureUnion], metaclass=ABCMeta):
+class FeatureUnionSparseFrames(
+    FeatureUnion,  # type:ignore
+):
+    """
+    FeatureUnion transformer that returns sparse data frames instead of arrays if one or
+    more of its transformers return a sparse data frame.
+    """
+
+    # noinspection PyPep8Naming
+    def _hstack(
+        self, Xs: List[Union[npt.NDArray[Any], sparse.spmatrix, pd.DataFrame]]
+    ) -> Union[npt.NDArray[Any], sparse.spmatrix, pd.DataFrame]:
+        stacked = hstack_frames(Xs)
+        return stacked if stacked is None else super()._hstack(Xs)
```

Comment:
I think `None` is missing here in the sum type that is returned. (As I understand `None` can be returned as `return stacked if stacked is None`)

Or shouldn't it be `return stacked if stacked is not None`?

Reply:
Yes it's a bug. Should be `if stacked is not None`. Will add a unit test.